### ISSUE TITLE
Lots of problems with links

### ISSUE THREAD

**Issue by mechanicalgux** [deanblackborough/php-quill-renderer#117]
Try this json:

```json
{"ops":[{"insert":"The "},{"attributes":{"italic":true,"link":"https://www.google.com"},"insert":"quick"},{"insert":" brown fox "},{"attributes":{"bold":true,"link":"https://www.google.com"},"insert":"jumps"},{"insert":" over t"},{"attributes":{"link":"https://www.google.com"},"insert":"he "},{"attributes":{"italic":true,"bold":true,"link":"https://www.google.com"},"insert":"lazy"},{"attributes":{"link":"https://www.google.com"},"insert":" do"},{"insert":"g... "},{"attributes":{"italic":true,"link":"https://www.amazon.com"},"insert":"Space"},{"insert":" "},{"attributes":{"bold":true,"link":"https://www.yahoo.com"},"insert":"removed"},{"insert":".\n"}]}
```

Links are ignored when another attribute like italic or bold text is applied, and space is removed between two words if both words have another attribute with the link.

**Comment by deanblackborough** [comment]
This is fixed in the v3.17.1 development branch. It has raised an issue though, turns out I never correctly fixed #109 so I need to fix that before I can release it.

I'm hoping to fix that tomorrow, I know what the problem is, it is just a little more involved than todays fix.

**Comment by mechanicalgux** [comment]
Thanks for the fix!
Still remains the space removed bug, try this:

```json
{"ops":[{"attributes":{"italic":true,"link":"https://www.amazon.com"},"insert":"Space"},{"insert":" "},{"attributes":{"bold":true,"link":"https://www.yahoo.com"},"insert":"removed"},{"insert":" but shoudn't"},{"attributes":{"bold":true},"insert":"."},{"insert":"\n"}]}
```

**Comment by deanblackborough** [comment]
Cheers @mechanicalgux, I'll work on new fix, v3.17.2 should be out before Monday
